### ISSUE TITLE
Combination of snap and steps seems to make keyboard non-responsive

### ISSUE THREAD

**Issue by souphan-adsk** [leongersen/noUiSlider#961]
When the `snap` and `range` (for custom steps) are enabled, the handles do not seem to respond to keyboard events.

This seems to be due to next/previous step prediction always being tied to calculated values as opposed to using the defined ones in this case.

**Comment by leongersen** [comment]
Could you add your configuration as an example?

**Comment by souphan-adsk** [comment]
Sure! The following seems enough to be able to reproduce the bug:
```javascript
let options = {
	start: 42,
	snap: true,
	range: {
		min: 0,
		'50%': 42,
		max: 1337,
	},
};
```

Without looking too much into the possible side effects or special cases, the following seems to be enough to fix the issue:

https://github.com/leongersen/noUiSlider/blob/547b3c33a3995f72727ae26f01ebcc721aef54c4/src/nouislider.js#L2254-L2259
```javascript
            // If snapped, directly use defined step value
            if (options.snap) {
                return [
                    value - nearbySteps.stepBefore.startValue || null,
                    nearbySteps.stepAfter.startValue - value || null
                ];
            }
...
```

**Comment by leongersen** [comment]
Thanks for the reproduction. I've just tried the proposed fix, and it looks good to me. The [`steps`](https://refreshless.com/nouislider/slider-values/#section-steps) method has two "special" cases:
- return `false` for a (sub-)range with no step value (doesn't apply for `snap`)
- return `null` for no possible previous/next step (case addressed)

Would you be interested in making a PR for this?

**Comment by souphan-adsk** [comment]
No problem. I'll have one ready in a bit.

**Comment by leongersen** [comment]
I've merged the PR and released noUiSlider 13.1.4 with this fix. Thanks for contributing, @souphan-adsk, much appreciated!
